spec/codebreaker_spec.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+require 'codebreaker/game'
+
+RSpec.describe 'Game' do
+  include Codebreaker
+  it 'Codebreaker need to be instant of Main' do
+    game = Codebreaker::Game.new
+    expect(game).to be_instance_of(Codebreaker::Game)
+  end
+  context '#play' do
+    let(:game) { Codebreaker::Game.new }
+    it 'Codebreaker game test play method' do
+      play_test('6543', '5643', '++--')
+      play_test('6543', '6411', '+-')
+      play_test('6543', '6544', '+++')
+      play_test('6543', '3456', '----')
+      play_test('6543', '6666', '+')
+      play_test('6543', '2666', '-')
+      play_test('6543', '2222', '')
+      play_test('6666', '1661', '++')
+      play_test('1234', '3124', '+---')
+      play_test('1234', '1524', '++-')
+      play_test('1234', '1234', '++++ (win)')
+    end
+    def play_test(user_code, secret_code, result)
+      game.new_game('Mykola', :medium)
+      expect(game.play(user_code, secret_code)).to eq(result)
+    end
```

Comment:
It would be better to set array and run each case in separate 'it' so u can see if several cases don't work
```
[all cases].each do |user_code, secret_code, result|
  it "when user code is #{user_code}, secret code is #{secret_code}, result is #{result}"
    test_itself
  end
end
```